File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/ledger/SubmitActionTest.java
package org.hyperledger.indy.sdk.ledger;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithPoolAndSingleWallet;
import org.hyperledger.indy.sdk.did.Did;
import org.hyperledger.indy.sdk.did.DidResults;
import org.json.JSONObject;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

public class SubmitActionTest extends IndyIntegrationTestWithPoolAndSingleWallet {

	private String did;

	@Before
	public void createDid() throws Exception {
		DidResults.CreateAndStoreMyDidResult result = Did.createAndStoreMyDid(wallet, "{}").get();
		did = result.getDid();
	}

	@Test
	public void testSubmitActionWorksForGetValidatorInfo() throws Exception {
		String getValidatorInfoRequest = Ledger.buildGetValidatorInfoRequest(did).get();
		getValidatorInfoRequest = Ledger.signRequest(wallet, did, getValidatorInfoRequest).get();
		Ledger.submitAction(pool, getValidatorInfoRequest, null, - 1).get();
	}

	@Test
	public void testSubmitActionWorksForPoolRestart() throws Exception {
		String poolRestartRequest = Ledger.buildPoolRestartRequest(did, "cancel", null).get();
		poolRestartRequest = Ledger.signRequest(wallet, did, poolRestartRequest).get();
		Ledger.submitAction(pool, poolRestartRequest, null, - 1).get();
	}

	@Test
	public void testSubmitActionWorksForNodes() throws Exception {
		String nodes = "[\"Node1\",\"Node2\"]";
		String getValidatorInfoRequest = Ledger.buildGetValidatorInfoRequest(did).get();
		getValidatorInfoRequest = Ledger.signRequest(wallet, did, getValidatorInfoRequest).get();
		String responseJson = Ledger.submitAction(pool, getValidatorInfoRequest, nodes, - 1).get();
		JSONObject response = new JSONObject(responseJson);
		assertEquals(2, response.length());
		assertTrue(response.has("Node1"));
		assertTrue(response.has("Node2"));
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/LedgerTests/SubmitActionTest.cs
using Hyperledger.Indy.DidApi;
using Hyperledger.Indy.LedgerApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json.Linq;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.LedgerTests
{
    [TestClass]
    public class SubmitActionTest : IndyIntegrationTestWithPoolAndSingleWallet
    {
        private string did;

        [TestInitialize]
        public async Task CreateDid()
        {
            var result = await Did.CreateAndStoreMyDidAsync(wallet, "{}");
            did = result.Did;
        }
        
        [TestMethod]
        public async Task TestSubmitActionWorksForGetValidatorInfo()
        {
            var getValidatorInfoRequest = await Ledger.BuildGetValidatorInfoRequestAsync(did);
            getValidatorInfoRequest = await Ledger.SignRequestAsync(wallet, did, getValidatorInfoRequest);
            await Ledger.SubmitActionAsync(pool, getValidatorInfoRequest, null, -1);
        }

        [TestMethod]
        public async Task TestSubmitActionWorksForPoolRestart()
        {
            var poolRestartRequest = await Ledger.BuildPoolRestartRequestAsync(did, "cancel", null);
            poolRestartRequest = await Ledger.SignRequestAsync(wallet, did, poolRestartRequest);
            await Ledger.SubmitActionAsync(pool, poolRestartRequest, null, -1);

        }

        [TestMethod]
        public async Task TestSubmitActionWorksForNodes()
        {
            var nodes = "[\"Node1\",\"Node2\"]";
            var getValidatorInfoRequest = await Ledger.BuildGetValidatorInfoRequestAsync(did);
            getValidatorInfoRequest = await Ledger.SignRequestAsync(wallet, did, getValidatorInfoRequest);
            var responseJson = await Ledger.SubmitActionAsync(pool, getValidatorInfoRequest, nodes, -1);
            var response = JObject.Parse(responseJson);
            Assert.AreEqual(2, response.Count);
            Assert.IsNotNull(response["Node1"]);
            Assert.IsNotNull(response["Node2"]);
        }
    }
}
